### PR TITLE
Creates move document upload UI [delivers #158768342]

### PR DESCRIPTION
**Pull request by stangah**
## Description

This builds off of #759. Now instead of supplying a `document_id` to the `createMoveDocument` endpoint, the client supplies a list of otherwise unassociated `upload_ids` which creates both the `Document` and `MoveDocument`.

## Code Review Verification Steps

* [x] All tests pass.
* [ ] Code follows the guidelines for [Logging](https://github.com/transcom/mymove/blob/master/docs/backend.md#logging)
* [ ] The requirements listed in
 [Querying the Database Safely](https://github.com/transcom/mymove/blob/master/docs/backend.md#querying-the-database-safely)
 have been satisfied.
* [ ] Any migrations/schema changes also update the diagram in docs/schema/dp3.sqs
* [ ] There are no aXe warnings for UI.
* [ ] This works in IE.
* Any new client dependencies (Google Analytics, hosted libraries, CDNs, etc) have been:
  * [ ] Communicated to @willowbl00
  * [ ] Added to the list of [network dependencies](https://github.com/transcom/mymove#client-network-dependencies)
* [ ] Request review from a member of a different team.
* [ ] (TRIAL) Have the Pivotal acceptance criteria been met for this change?

## References

* [Pivotal story](https://www.pivotaltracker.com/story/show/158768342) for this change

## Screenshots

move document upload

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -181,7 +181,7 @@ db_e2e_init: tools_build db_dev_run db_test_reset
 	DB_HOST=localhost DB_PORT=5432 DB_NAME=test_db \
 		./bin/soda -e test migrate -c config/database.yml -p cypress/migrations up
 
-db_e2e_reset: tools_build db_dev_run
+db_e2e_reset: db_dev_run
```
Comment (stangah) [inline Makefile]: This should only ever run inside our e2e test, so rebuilding tools isn't necessary

File: cypress/integration/office/documentViewer.js [hunk 2]
```diff
@@ -8,19 +8,38 @@ describe('The document viewer', function() {
...
+      cy.contains('Upload a new document');
+      cy.get('button.submit').should('be.disabled');
+      cy.get('input[name="title"]').type('super secret info document');
+      cy.get('select[name="move_document_type"]').select('Other document type');
+      cy.get('input[name="notes"]').type('burn after reading');
+      cy.get('button.submit').should('be.disabled');
+
+      cy.upload_file('.filepond--root', 'top-secret.png');
+      cy
+        .get('button.submit')
```
Comment (stangah) [inline cypress/integration/office/documentViewer.js]: The upload takes a bit, but Cypress has built-in timeouts for these actions. I'm relying on the fact that the upload should finish before our timeout here. It always worked for me, but curious if there's a better way to handle this.

File: pkg/models/move.go [hunk 3]
```diff
@@ -171,27 +171,65 @@ func FetchMove(db *pop.Connection, session *auth.Session, id uuid.UUID) (*Move,
...
-	if err != nil || verrs.HasAny() {
-		return nil, verrs, err
-	}
+	db.Transaction(func(db *pop.Connection) error {
+		transactionError := errors.New("Rollback The transaction")
+
+		// Make a generic Document
+		newDoc := Document{
+			ServiceMemberID: m.Orders.ServiceMemberID,
+			Name:            "this field is useless",
```
Comment (amitch23) [inline pkg/models/move.go]: Should this field be removed from the documents table?

File: pkg/handlers/move_documents.go [hunk 4]
```diff
@@ -47,16 +47,22 @@ func (h CreateMoveDocumentHandler) Handle(params moveop.CreateMoveDocumentParams
...
+			return responseForError(h.logger, err)
+		}
+		uploads = append(uploads, upload)
 	}
 
 	newMoveDocument, verrs, err := move.CreateMoveDocument(h.db,
-		document,
+		uploads,
 		models.MoveDocumentType(payload.MoveDocumentType),
+		*payload.Title,
```
Comment (amitch23) [inline pkg/handlers/move_documents.go]: Can you add the title to the payloadForMoveDocumentModel above?

File: src/scenes/Office/DocumentViewer/index.jsx [hunk 5]
```diff
@@ -89,7 +100,14 @@ class DocumentViewer extends Component {
               />
               <PrivateRoute
                 path={listUrl}
-                render={() => <div> list coming soon</div>}
+                render={() => (
+                  <Fragment>
+                    <Link to={newUrl} moveId={move.id}>
```
Comment (amitch23) [inline src/scenes/Office/DocumentViewer/index.jsx]: The browser console is complaining about the `moveId` prop. I changed it to lower case, `moveid` and it's happy, and then I removed it and things seem to work. Perhaps it's not needed?

File: pkg/models/upload.go [hunk 6]
```diff
@@ -75,7 +75,7 @@ func FetchUpload(db *pop.Connection, session *auth.Session, id uuid.UUID) (Uploa
 			return Upload{}, docErr
 		}
 	} else if upload.UploaderID != session.UserID {
-		return Upload{}, ErrFetchNotFound
+		return Upload{}, errors.Wrap(ErrFetchNotFound, "user ID doesn't match uploader ID")
```
[the remark this answers is not in the document]
Comment (stangah) [reply pkg/models/upload.go]: This `else` case only happens if there's no `document` attached, which should only happen when you are in the process of uploading files and saving them to a document

**Pull request merged**